packages/appsync-modelgen-plugin/src/__tests__/visitors/appsync-swift-visitor.test.ts

```diff
@@ -2976,4 +2976,38 @@ describe('AppSyncSwiftVisitor', () => {
       expect(generatedMetaComment).toMatchSnapshot();
     });
   });
+  describe('custom references', () => {
+    it('sets the association to the references field for hasOne/belongsTo', () => {
+      const schema = /* GraphQL */ `
+        type Project @model {
+          projectId: ID! @primaryKey(sortKeyFields: ["name"])
+          name: String!
+          team: Team @hasOne(references: ["projectId"])
```

Comment:
```suggestion
          team: Team @hasOne(references: ["projectId", "projectName"])
```

Reply:
The sort key defined on the Primary model `Project` needs to be explicitly defined the Related model `Team` as a property and included in references argument.